### PR TITLE
eslint@4.5.x

### PR DESCRIPTION
**Pull request by davidchambers**
I'll add links to the various new rules to make this pull request easier to review. :)

### CODE REVIEW

File: eslint-common.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   "dot-location": ["property"],
   "eol-last": ["always"],
   "eqeqeq": ["always", {"null": "never"}],
+  "for-direction": [],
```
Comment (davidchambers) [inline eslint-common.json]: <https://eslint.org/docs/rules/for-direction>

File: eslint-common.json [hunk 2]
```diff
@@ -63,20 +64,24 @@
   "no-useless-escape": [],
   "no-whitespace-before-property": [],
   "no-with": [],
+  "object-curly-newline": [{"consistent": true}],
```
Comment (davidchambers) [inline eslint-common.json]: <https://eslint.org/docs/rules/object-curly-newline>

File: eslint-common.json [hunk 3]
```diff
@@ -63,20 +64,24 @@
...
   "object-curly-spacing": ["never"],
   "object-property-newline": [{"allowMultiplePropertiesPerLine": true}],
   "operator-assignment": ["always"],
+  "padded-blocks": ["never"],
```
Comment (davidchambers) [inline eslint-common.json]: <https://eslint.org/docs/rules/padded-blocks>

File: eslint-common.json [hunk 4]
```diff
@@ -63,20 +64,24 @@
...
   "quotes": ["single", {"avoidEscape": true}],
   "radix": ["always"],
   "semi": ["always", {"omitLastInOneLineBlock": false}],
   "semi-spacing": [{"before": false, "after": true}],
+  "semi-style": ["last"],
```
Comment (davidchambers) [inline eslint-common.json]: <https://eslint.org/docs/rules/semi-style>

File: eslint-common.json [hunk 5]
```diff
@@ -63,20 +64,24 @@
...
   "space-before-blocks": ["always"],
   "space-before-function-paren": [{"anonymous": "never", "named": "never"}],
   "space-in-parens": ["never", {"exceptions": []}],
   "space-infix-ops": [],
   "space-unary-ops": [{"words": true, "nonwords": false, "overrides": {}}],
   "spaced-comment": ["always", {"line": {"markers": ["#", "."], "exceptions": ["."]}}],
   "strict": ["safe"],
+  "switch-colon-spacing": [{"before": false, "after": true}],
```
Comment (davidchambers) [inline eslint-common.json]: <https://eslint.org/docs/rules/switch-colon-spacing>

File: eslint-common.json (hunk 1 above)
Comment (Avaq) [inline eslint-common.json]: Just out of curiosity, why disable this rule in `common`, and enable it in `es3` and `es6`?

Comment (davidchambers) [inline eslint-common.json]: That's not actually the case:

```make
eslint-%.json: eslint-%-only.json eslint-common.json scripts/eslint-combine
	scripts/eslint-combine eslint-common.json '$<' >'$@'
```

Running `make` updates both derived files based on changes to __eslint-common.json__. The script which combines the common rules and the ES3/ES6 rules sets the level to `"error"` in all cases, to save us from having to specify it for every rule. :)

Comment (Avaq) [inline eslint-common.json]: Ah, I see. :)

I'm [using](https://github.com/wearereasonablepeople/eslint-config-warp/blob/master/es6.js#L5) eslint's built-in extension capability to share rules between configurations in the config I maintain, as opposed to a build process. Any reason to favour the build option?

Comment (davidchambers) [inline eslint-common.json]: > Any reason to favour the build option?

This project predates ESLint's extension mechanism, which was added fairly recently. I'll investigate the possibility of using `extends` here. Thanks for the suggestion!

File: eslint-common.json (hunk 3 above)
Comment (davidchambers) [inline eslint-common.json]: I removed this rule when I discovered that it flags this code:

```javascript
  (function(f) {
>>
    'use strict';

    /* istanbul ignore else */
    if (typeof module === 'object' && typeof module.exports === 'object') {
      module.exports = f(require('sanctuary-type-classes'));
    } else if (typeof define === 'function' && define.amd != null) {
      define(['sanctuary-type-classes'], f);
    } else {
      self.sanctuaryIdentity = f(self.sanctuaryTypeClasses);
    }
>>
  }(function(Z) {
>>
    'use strict';

    ...
>>
  }));
```

File: eslint-common.json (hunk 1 above)
Comment (davidchambers) [reply eslint-common.json]: We're now using `extends`. How do the changes look, Aldwin?

**Pull request merged**